File: algorithm/Stack/queueWithStack/queueWithStack.py
class Queue:
  #initialize your data structure here.
  def __init__(self):
    self.inbox  = []
    self.outbox = []

  #@param x, an integer
  #@return nothing
  def push(self,x):
    self.inbox.append(x)

  #@return nothing
  def pop(self):
    if len(self.outbox) == 0:
      while len(self.inbox) > 0:
        self.outbox.append(self.inbox.pop())
      self.outbox.pop()

  #@return an integer
  def peek(self):
    if len(self.outbox) == 0:
      while len(self.inbox) > 0:
          self.outbox.append(self.inbox.pop())
    return self.outbox[len(self.outbox)-1]

  #@return an boolean
  def empty(self):
    return len(self.inbox) == 0 and len(self.outbox) == 0

File: algorithm/Stack/queueWithStack/queueWithStack.java
public class queueWithStack {

  Stack<Integer> inbox  = new Stack<Integer>();
  Stack<Integer> outbox = new Stack<Integer>();

  //push element x to the back of queue
  public void push(int x) {
    inbox.push(x);
  }

  //removes the element from in front of queue
  public void pop() {
    if (outbox.isEmpty()) {
      while(!inbox.isEmpty()){
        outbox.push(inbox.pop());
      }
    }
    outbox.pop();
  }

  //get the front element.
  public int peek() {
    if (outbox.isEmpty()){
      while(!inbox.isEmpty()){
        outbox.push(inbox.pop());
      }
    }
    outbox.peek();
  }

  //return whether the queue is empty.
  public boolean empty() {
    return inbox.isEmpty() && outbox.isEmpty();
  }

}
